src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -320,6 +320,13 @@ VtolAttitudeControl::Run()
 		_airspeed_validated_sub.update(&_airspeed_validated);
 		_tecs_status_sub.update(&_tecs_status);
 		_land_detected_sub.update(&_land_detected);
+
+		if (_home_position_sub.updated()) {
+			home_position_s home_position{};
+			_home_position_sub.copy(&home_position);
```

Comment:
```suggestion
			if (_home_position_sub.copy(&home_position) && home_position.valid_alt) {
				_home_position_z = home_position.z;
			} else {
				_home_position_z = NAN;
			}
```

Then you can also skip zeroing `home_position`.

Reply:
done in e1247a85c2967118f6bae5f73c0cdb3529fc2b32